src/FSharpPlus/Data/NonEmptyList.fs

```diff
@@ -35,15 +35,32 @@ type NonEmptyList<'t> = {Head: 't; Tail: 't list} with
 /// Basic operations on NonEmptyList
 [<RequireQualifiedAccess>]
 module NonEmptyList =
+    /// <summary>Builds a non empty list.</summary>
     let create x xs = {Head = x; Tail = xs}
+    /// <summary>Builds a non empty list with a single element.</summary>
     let singleton x = {Head = x; Tail = []}
+    /// <summary>Builds a list from the given non empty list.</summary>
     let toList {Head = x; Tail = xs} = x::xs
+    /// <summary>Builds a sequence from the given non empty list.</summary>
     let toSeq  {Head = x; Tail = xs} = seq { yield x; yield! xs; }
+    /// <summary>Builds an array from the given non empty list.</summary>
     let toArray nel = toList nel |> List.toArray
-    let ofArray (array : _ array) = match Array.tryHead array with | Some head -> create head (Array.tail array |> Array.toList) | _ -> failwith "cannot construct empty non empty list"
-    let ofList (list : _ list) = match List.tryHead list with | Some head -> create head (List.tail list) | _ -> failwith "cannot construct empty non empty list"
-    let ofSeq (list : _ seq) = match Seq.tryHead list with | Some head -> create head (Seq.tail list |> Seq.toList) | _ -> failwith "cannot construct empty non empty list"
+    /// <summary>Builds a non empty list from the given array.</summary>
+    /// <param name="array">The input array.</param>
+    /// <returns>Non empty list containing the elements of the array.</returns>
+    /// <exception cref="System.ArgumentException">Thrown when the input array is empty.</exception>
+    /// <remarks>Throws exception for empty array</remarks>
+    let ofArray (array : _ array) = match Array.tryHead array with | Some head -> create head (Array.tail array |> Array.toList) | _ -> invalidArg "array" "The input array was empty."
```

Comment:
Can the use of tryHead be removed seems redundant as the same fail is raised in Array.head

Reply:
It might be better to use `Array.isEmpty` and `array.[0]`.